atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/AdjustableSweepBatchConfigSourceTest.java

```diff
@@ -0,0 +1,216 @@
+            // When
+            adjustableConfig.increaseMultiplier();
+
+            // Then
+            batchSizeMultiplierIncreases();
+            batchSizeMultiplierDoesNotExceedOne();
+
+            maxCellTsPairsToExamineDoesNotDecrease();
+            candidateBatchSizeDoesNotDecrease();
+            deleteBatchSizeDoesNotDecrease();
+
+            maxCellTsPairsToExamineDoesNotExceedBaseConfig();
+            candidateBatchSizeDoesNotExceedBaseConfig();
+            deleteBatchSizeDoesNotExceedBaseConfig();
+
+            updatePreviousValues();
+        }
+    }
+
+    private void whenDecreasingTheMultiplier_thenAdjustedConfigValuesDecrease() {
+        for (int i = 0; i < 10_000; i++) {
+            // When
+            adjustableConfig.decreaseMultiplier();
+
+            // Then
+            batchSizeMultiplierDecreases();
+            maxCellTsPairsToExamineDecreasesToAMinimumOfOne();
+            candidateBatchSizeDecreasesToAMinimumOfOne();
+            deleteBatchSizeDecreasesToAMinimumOfOne();
+
+            updatePreviousValues();
+        }
+    }
+
+    private void configWithValues(int maxCellTsPairsToExamine, int candidateBatchSize, int deleteBatchSize) {
+        adjustableConfig = AdjustableSweepBatchConfigSource.create(() -> new SweepBatchConfig() {
+            @Override
+            public int maxCellTsPairsToExamine() {
+                return maxCellTsPairsToExamine;
+            }
+
+            @Override
+            public int candidateBatchSize() {
+                return candidateBatchSize;
+            }
+
+            @Override
+            public int deleteBatchSize() {
+                return deleteBatchSize;
+            }
+        });
+
+        updatePreviousValues();
+    }
+
+    private void batchSizeMultiplierDecreases() {
+        assertThat(adjustableConfig.getBatchSizeMultiplier(), is(lessThanOrEqualTo(previousMultiplier)));
+    }
+
+    private void maxCellTsPairsToExamineDecreasesToAMinimumOfOne() {
+        int newValue = adjustableConfig.getAdjustedSweepConfig().maxCellTsPairsToExamine();
+        if (newValue == 1) {
+            return;
+        }
+        assertThat(newValue, is(lessThan(previousMaxCellTsPairsToExamine)));
+    }
+
+    private void candidateBatchSizeDecreasesToAMinimumOfOne() {
+        int newValue = adjustableConfig.getAdjustedSweepConfig().candidateBatchSize();
+        if (newValue == 1) {
+            return;
+        }
+        assertThat(newValue, is(lessThan(previousCandidateBatchSize)));
+    }
+
+    private void deleteBatchSizeDecreasesToAMinimumOfOne() {
+        int newValue = adjustableConfig.getAdjustedSweepConfig().deleteBatchSize();
+        if (newValue == 1) {
+            return;
+        }
+        assertThat(newValue, is(lessThan(previousDeleteBatchSize)));
+    }
+
+    private void batchSizeMultiplierIncreases() {
+        assertThat(adjustableConfig.getBatchSizeMultiplier(), is(greaterThanOrEqualTo(previousMultiplier)));
+    }
```

Comment:
nit: This seems like a `...DoesNotDecrease()`, following consistency with the method for max-cell-ts-pairs. I'm guessing we need the equality case for when it's 1 - in which case we could do `increasesToAMaximumOfOne()` and something analogous to the batch sizes needing to be at least 1.

Reply:
Refactored this to be clearer